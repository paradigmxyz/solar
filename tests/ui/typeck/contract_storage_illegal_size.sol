//@ compile-flags: -Zprint-max-storage-sizes

// Okay because 2^256 - 1 slots, which are the maximum permissible slots, are used
contract AlmostIllegal { //~ NOTE: :AlmostIllegal requires a maximum of 115792089237316195423570985008687907853269984665640564039457584007913129639935 storage slots
    uint256[115792089237316195423570985008687907853269984665640564039457584007913129639935]
        public x;
}

// Not Okay because 2^256 slots are used
contract Illegal { //~ ERROR: contract requires too much storage
    uint256[115792089237316195423570985008687907853269984665640564039457584007913129639935]
        public x;
    uint256 y;
}
