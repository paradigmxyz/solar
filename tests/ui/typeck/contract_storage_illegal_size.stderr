note: ROOT/tests/ui/typeck/contract_storage_illegal_size.sol:AlmostIllegal requires a maximum of 115792089237316195423570985008687907853269984665640564039457584007913129639935 storage slots
  --> ROOT/tests/ui/typeck/contract_storage_illegal_size.sol:LL:CC
   |
LL | contract AlmostIllegal {
   |          ^^^^^^^^^^^^^

error: contract requires too much storage
  --> ROOT/tests/ui/typeck/contract_storage_illegal_size.sol:LL:CC
   |
LL | contract Illegal {
   |          ^^^^^^^

error: aborting due to 1 previous error; 1 note emitted

